File: 409. Longest Palindrome/Python/Solution.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        count_array = [0] * (ord('z') - ord('A') + 1)
        pair = 0
        is_odd_exists = False
        for item in s:
            count_array[ord(item) - ord('A')] += 1
        for item in count_array:
            pair += item // 2
            if not is_odd_exists:
                is_odd_exists = item % 2 != 0
        return pair * 2 + (1 if is_odd_exists else 0)

File: 409. Longest Palindrome/C++/Solution.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int countArray['z' - 'A' + 1] = {0};
        int pair = 0;
        bool isOddExists = false;
        for (char item : s) {
            countArray[item - 'A']++;
        }
        for (int item : countArray) {
            pair += item / 2;
            if (!isOddExists) {
                isOddExists = item % 2 == 1;
            }
        }
        return pair * 2 + (isOddExists ? 1 : 0);
    }
};